src/migrations/20250116140614-add-activity-report-objective-citations.js

```diff
@@ -0,0 +1,47 @@
+const { prepMigration } = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+      // Create ActivityReportObjectiveCitations table.
+      await queryInterface.createTable('ActivityReportObjectiveCitations', {
+        id: {
+          type: Sequelize.INTEGER,
+          autoIncrement: true,
+          primaryKey: true,
+        },
+        activityReportObjectiveId: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+        },
+        citation: {
+          type: Sequelize.TEXT,
+          allowNull: false,
+        },
+        monitoringReferences: {
```

Comment:
Putting our links in a jsonb seems like a hair-raising choice. Can you add an explanation (could be a comment, or a comment linking to a discussion) of why we're doing this rather than using regular columns or link tables for which we can enforce referential integrity?

Reply:
@hardwarehuman the time and complexity of joining to the monitoring data each time is something we want to avoid. This approach allows us to store all relevant information at the grant level so it can be quickly displayed on the UI or looked up (via joins)  if needed.

Reply:
@AdamAdHocTeam Is the idea that we're storing the referential data but don't intend to ever use it? For example, if the referenced records change or are superseded at some point in the future, are we fine continuing to refer to stale data because we only want a snapshot?

Or did we make this decision mainly because we don't need to use the referenced records live, but we do want to keep the references consistent over time? If that is the case, then we probably want to add some future work to either break these out into traditional structures or use some other mechanism to ensure we remain consistent.

Reply:
@hardwarehuman The main reason for this is that it is not guarantied to be a single match. One citation may come through multiple findings (multiple findingIds).

Reply:
@hardwarehuman I added a comment.